src/frequenz/client/dispatch/_types.py

```diff
@@ -0,0 +1,442 @@
+    """Indicates if the dispatch is a dry run.
+
+    Executed for logging and monitoring without affecting actual component states."""
+
+    payload: dict[str, Any]
+    """The dispatch payload containing arbitrary data.
+
+    It is structured as needed for the dispatch operation."""
+
+    recurrence: RecurrenceRule
+    """The recurrence rule for the dispatch.
+
+    Defining any repeating patterns or schedules."""
+
+    @classmethod
+    def from_dispatch(cls, dispatch: "Dispatch") -> "DispatchCreateRequest":
+        """Convert a dispatch to a dispatch create request.
+
+        Args:
+            dispatch: The dispatch to convert.
+
+        Returns:
+            The converted dispatch create request.
+        """
+        return DispatchCreateRequest(
+            microgrid_id=dispatch.microgrid_id,
+            type=dispatch.type,
+            start_time=dispatch.start_time,
+            duration=dispatch.duration,
+            selector=dispatch.selector,
+            is_active=dispatch.is_active,
+            is_dry_run=dispatch.is_dry_run,
+            payload=dispatch.payload,
+            recurrence=dispatch.recurrence,
+        )
+
+    @classmethod
+    def from_protobuf(
+        cls, pb_object: PBDispatchCreateRequest
+    ) -> "DispatchCreateRequest":
+        """Convert a protobuf dispatch create request to a dispatch.
+
+        Args:
+            pb_object: The protobuf dispatch create request to convert.
+
+        Returns:
+            The converted dispatch.
+        """
+        return DispatchCreateRequest(
+            microgrid_id=pb_object.microgrid_id,
+            type=pb_object.type,
+            start_time=pb_object.start_time.ToDatetime().replace(tzinfo=timezone.utc),
+            duration=timedelta(seconds=pb_object.duration),
+            selector=_component_selector_from_protobuf(pb_object.selector),
+            is_active=pb_object.is_active,
+            is_dry_run=pb_object.is_dry_run,
+            payload=MessageToDict(pb_object.payload),
+            recurrence=RecurrenceRule.from_protobuf(pb_object.recurrence),
+        )
+
+    def to_protobuf(self) -> PBDispatchCreateRequest:
+        """Convert a dispatch to a protobuf dispatch create request.
+
+        Returns:
+            The converted protobuf dispatch create request.
+        """
+        pb_request = PBDispatchCreateRequest()
+
+        pb_request.microgrid_id = self.microgrid_id
+        pb_request.type = self.type
+        pb_request.start_time.FromDatetime(self.start_time)
+        pb_request.duration = int(self.duration.total_seconds())
+        pb_request.selector.CopyFrom(_component_selector_to_protobuf(self.selector))
+        pb_request.is_active = self.is_active
+        pb_request.is_dry_run = self.is_dry_run
+        pb_request.payload.update(self.payload)
+        pb_request.recurrence.CopyFrom(self.recurrence.to_protobuf())
+
+        return pb_request
+
+
+@dataclass(kw_only=True)
+class Dispatch(DispatchCreateRequest):
```

Comment:
I wouldn't do this, besides both could diverge, this means someone could accidentally pass a `Dispatch` object where a `DispatchCreateRequest` is expected. I know it should be unlikely because `DispatchCreateRequest` should only be used internally, but that's actually another issue, we should better separate classes that are supposed to be public from classes that should be private, and public classes shouldn't inherit from private classes as users shouldn't have access to the private class.

Reply:
I really hate the repetition that that will cause...

Reply:
Again, this is supposed to be stable, is a one time thing.